ci/test/lint-deps/x86_64-unknown-linux-gnu-no-default-features

```diff
@@ -1 +1,8 @@
 # generated by ci/test/lint-deps.sh -- see ci/test/lint-deps/README.md for details
+tikv_jemalloc_ctl
```

Comment:
I have no idea if this is ok or not. Can someone who knows comment?

Reply:
No, that's incorrect. No default features should not pull in jenalloc.

Reply:
My read of envd's current Cargo.toml on main is that it specifically does pull it in?

https://github.com/MaterializeInc/materialize/blob/bacf255163a96426fa2bafe0689b4616dc58a004/src/environmentd/Cargo.toml#L144

This change adds 4 optional dependencies that are only enabled by a non-default feature. Am I reading something wrong? Is envd a weirdo exception because it's a binary and not a lib?